### PR TITLE
feat(vector): :sparkles: add envFrom support

### LINKED ISSUE #214: vector _pod.tpl does not offer envFrom
We have an existing `Secret` that would be very handy for providing to vector just like we provide it to other workloads, but the [current `_pod.tpl`](https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/templates/_pod.tpl) only accepts `.Values.env`, wherein we have to unpack the keys using `secretKeyRef`

I wanted to bring that missing `envFrom` to your attention

### PR DESCRIPTION
**Pull request by dannylongeuay**
Closes #214

### CODE REVIEW

**Comment by dannylongeuay** [comment]
@spencergilbert do you have any cycles to review this?

**Comment by spencergilbert** [comment]
👍 yep, should be able to get to it this week

File: charts/vector/values.yaml [hunk 1]
```diff
@@ -152,6 +152,11 @@ env: []
   #       name: vector
   #       key: awsAccessKeyId
 
+# envFrom -- Define all Secret's data as container environment variables
```
Comment (spencergilbert) [inline charts/vector/values.yaml:155]:
```suggestion
# envFrom -- Define environment variables from Secrets or ConfigMaps
```

File: charts/vector/examples/datadog-values.yaml [hunk 2]
```diff
@@ -35,6 +35,11 @@ env:
   - name: VECTOR_LOG_FORMAT
     value: json
 
+# envFrom -- Define all Secret's data as container environment variables
```
Comment (spencergilbert) [inline charts/vector/examples/datadog-values.yaml:38]:
```suggestion
# envFrom -- Define environment variables from Secrets or ConfigMaps
```

File: charts/vector/README.md [hunk 3]
```diff
@@ -140,6 +140,7 @@ helm install --name <RELEASE_NAME> \
 | dnsConfig | object | `{}` | Specify DNS configuration options for Vector Pods # Ref: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config |
 | dnsPolicy | string | `"ClusterFirst"` | Specify DNS policy for Vector Pods # Ref: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy |
 | env | list | `[]` | Set environment variables in Vector containers # The examples below leverage examples from secrets.generic and assume no name overrides with a Release name of "vector" |
+| envFrom | list | `[]` | Define all Secret's data as container environment variables |
```
Comment (spencergilbert) [inline charts/vector/README.md:143]:
```suggestion
| envFrom | list | `[]` | Define environment variables from Secrets or ConfigMaps |
```

**Review by spencergilbert — APPROVED** [review]
👍 CI works! Thanks for the contribution, I tweaked the comment/doc slightly since `envFrom` can be used with either Secrets or ConfigMaps and I wanted to make sure that was clear - what do you think?

File: charts/vector/values.yaml (hunk 1 above)
Comment (dannylongeuay) [reply charts/vector/values.yaml:155]: good clarification

**Comment by dannylongeuay** [comment]
> +1 CI works! Thanks for the contribution, I tweaked the comment/doc slightly since `envFrom` can be used with either Secrets or ConfigMaps and I wanted to make sure that was clear - what do you think?

No problem, this will make my life much easier! I like the change and commited it!

**Comment by spencergilbert** [comment]
I wish the "approve and run ci" button just worked for the entire PR, and not on every change 😭 

Thanks again for the contribution @dannylongeuay!

**Pull request merged**